torchdynamo/variables/tensor.py

```diff
@@ -419,7 +420,16 @@ def call_method(
             sizes = [variables.ConstantVariable(x) for x in self.size]
             constant_result = SizeVariable(sizes, **options)
         elif name == "numel" and self.size is not None:
-            constant_result = ConstantVariable(product(self.size), **options)
+            if config.capture_scalar_outputs:
```

Comment:
This path is only hit in static shape mode, so the number of elements is a constant and guarded on.